src/type-layout.md

```diff
@@ -32,6 +32,28 @@ as [dynamically sized types]. Since all values of a `Sized` type share the same
 size and alignment, we refer to those shared values as the size of the type and
 the alignment of the type respectively.
 
+### Minimum Size and Alignment
+
+Every type, including [dynamically sized types], has a "minimum size" and "minimum
+alignment." Every value of a type must have a size at least as large as the type's
+minimum size and an alignment at least as large as the type's minimum alignment.
+Some notable cases of minimum size and alignment are:
+- For [`Sized`] types, the type's minimum size and minimum alignment
+  are always equal to the type's size and alignment, respectively
+- For a [slice type](#slice-layout), `[T]`, the minimum size is 0 bytes (corresponding
+  to a slice with 0 elements), and the minimum alignment is the alignment of `T`
+- For a [trait object](#trait-object-layout), the minimum size is 0 and the minimum
+  alignment is 1
+- For a struct type with a dynamically-sized field, the minimum size is taken to be
+  the size of the struct when the dynamically-sized field has *its* minimum size.
+  The struct's minimum alignment is taken to be the alignment of the struct when the
+  dynamically-sized field has *its* minimum alignment. When relying on this property
+  of dynamically-sized structs, be careful not to assume too much! For example,
+  this property provides *no* guarantees about [`repr(Rust)`](#the-rust-representation)
+  types, as such a type can have arbitrarily large size and alignment regardless of
+  the sizes and alignments of its fields
+- Every type's minimum size is less than or equal to `isize::MAX`
```

Comment:
This bullet doesn't belong into the list of "notable cases of minimum size and alignment are", it belongs before or after the list. Also it should say what happens when one declares a type whose minimum size exceeds this limit.

Reply:
> This bullet doesn't belong into the list of "notable cases of minimum size and alignment are", it belongs before or after the list.

Done.

> Also it should say what happens when one declares a type whose minimum size exceeds this limit.

Do we need to specify that? Presumably the implication is just that such a program would not conform to the spec, and so we couldn't permit it to compile, but a) that's a logical implication of the text here and, b) I don't believe we need to specify the *manner in which* it fails to compile.

I'm happy to also specify that such programs won't compile - I just want to make sure I'm understanding correctly that this is just a nice-to-have rather than something that affects the semantics of this text.

Reply:
Sounds reasonable. Presumably resolving https://github.com/rust-lang/reference/pull/1482#issuecomment-2036255754 is required in order for us to agree on a wording for this?

Reply:
The questions of "which types are checked" and "what happens when a type fails the check" seem orthogonal to me, so I don't see an overlap with the other discussion.